### PR TITLE
Make test suite setup and teardown functions static

### PR DESCRIPTION
**Pull request by mbknust**
In order to migrate to pw_unit_test, `SetUpTestSuite` and `TearDownTestSuite` ought to be static member functions of a test fixture class.

But nltest still expects them to take a `void *` argument and return an int, so I wrap them with [macros](https://github.com/mbknust/connectedhomeip/blob/487741605af00e5420c260bcf01c7ffdaa638288/src/lib/support/UnitTestRegistration.h#L51-L63) each time they're used inside a `nlTestSuite` declaration.

**Followup**: Remove `NL_TEST_WRAP_FUNCTION` and `NL_TEST_WRAP_METHOD`, when they are no longer needed.

These changes used to be part of #33002, but that PR was split into smaller pieces.

### CODE REVIEW

File: src/transport/tests/LoopbackTransportManager.h [hunk 1]
```diff
@@ -27,26 +27,26 @@ class LoopbackTransportManager
 {
 public:
     /// Initialize the underlying layers.
-    CHIP_ERROR Init()
+    static CHIP_ERROR Init()
```
Comment (arkq) [inline src/transport/tests/LoopbackTransportManager.h:30]: Could you change the name of this function, if it needs to be static. By convention ([almost all](https://github.com/search?q=repo%3Aproject-chip%2Fconnectedhomeip+%22static+CHIP_ERROR+Init%28%29%22&type=code)) `Init()` functions are not static in whole project. Making this `Init()` as static might confuse someone...

Comment (mbknust) [reply src/transport/tests/LoopbackTransportManager.h]: Fixed; LoopbackTransportManager is no longer a singleton, instead it's a static member of LoopbackMessagingContext

**Review by arkq — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by feasel0 — APPROVED** [review]
This looks good.

File: src/transport/tests/UDPTransportManager.h [hunk 2]
```diff
@@ -29,33 +29,33 @@ class UDPTransportManager
 {
 public:
     /// Initialize the underlying layers.
-    CHIP_ERROR Init()
+    static CHIP_ERROR Init()
```
[the remark this answers is not in the document]
Comment (mbknust) [reply src/transport/tests/UDPTransportManager.h]: I changed this code to be written analogously to `LoopbackTransportManager`

**Pull request merged**